pkg/api/types.go

```diff
@@ -1015,6 +1015,21 @@ func (p *Properties) HasZonesForAllAgentPools() bool {
 	return false
 }
 
+// IsUbuntuDistroForAllAgentPools returns true if all of the agent pools plus masters are running the base Ubuntu image
+func (p *Properties) IsUbuntuDistroForAllAgentPools() bool {
```

Comment:
Function name doesn't accurately reflect what it's actually doing since it implies the check is for agent pools only, should be `HasNonUbuntuDistroProfiles` (then reverse the true/false returned), `IsUbuntuDistroForAllNodes`  or something like that.

Reply:
done